### ISSUE TITLE
Setting up Cakeshop for pre-existing Quorum network with Istanbul consensus

### ISSUE THREAD

**Issue by vasa-develop** [ConsenSys/cakeshop#166]
#### Description of problem
I have set up a 5 node Quorum Network with Istanbul consensus using [this guide](https://github.com/jpmorganchase/quorum/blob/50429f8cd3d11e045dd00a5c6d0b8eaa37072155/docs/Getting%20Started/Creating-A-Network-From-Scratch.md).

I'm trying to setup cakeshop (v0.11.0 Release Candidate 4) by attaching it with one of the Quorum nodes to monitor this network.

#### Node details
geth.url=http\://localhost\:22000
geth.networkid=10
geth.node.port=30303
geth.consensus.mode: istanbul
geth.istanbul.url=??? (**WHAT IS THIS?**)

I need to make changes to the `application.properties` accordingly. 

**Please suggest the changes that I need to add to the below config**

Let me know if you need any other additional details.

```
#Thu Mar 26 07:33:57 UTC 2020
cakeshop.database.vendor=hsqldb
cakeshop.hibernate.hbm2ddl.auto=update
cakeshop.hibernate.jdbc.batch_size=20
cakeshop.jdbc.pass=sdk
cakeshop.jdbc.user=sdk
cakeshop.mvc.async.pool.queue.max=2000
cakeshop.mvc.async.pool.threads.core=250
cakeshop.mvc.async.pool.threads.max=1000
contract.poll.delay.millis=5000
contract.registry.addr=
endpoints.actuator.enabled=true
geth.auto.start=false
geth.auto.stop=false
geth.bootnode.address=
geth.bootnode.key=
geth.bootnodes.list=
geth.consensus.mode=istanbul
geth.cors.enabled=false
geth.cors.url=
geth.cred1=admin
geth.cred2=$2a$10$dbGiTnfK/w8MhcpIj3XgROYXRsFMlEYJRWoUYArkr8aSPypUFV25G
geth.datadir=/home/ubuntu/data/local/ethereum
geth.db.enabled=true
geth.identity=ubuntu
geth.istanbul.url=
geth.log=/logs
geth.mining=true
geth.networkid=10
geth.node.port=30303
geth.params.extra=
geth.raft.blocktime=100
geth.raft.network.id=
geth.raft.port=50401
geth.release.url=
geth.rpcapi.list=admin,db,eth,debug,miner,net,shh,txpool,personal,web3
geth.startup.mode=standalone
geth.tools.url=
geth.transaction_manager.peers=http\://localhost\:9102
geth.transaction_manager.third_party_url=http\://localhost\:9202
geth.transaction_manager.url=http\://localhost\:9102
geth.unlock.timeout=5000
geth.url=http\://localhost\:22000
geth.verbosity=
geth.vote.contract.addr=0x0000000000000000000000000000000000000020
log4j.rootLogger=DEBUG, stdout
management.context-path=/manage
management.security.enabled=false
nodejs.binary=node
security.basic.enabled=false
security.ignored=/**
server.compression.enabled=true
server.compression.mime-types=application/json,application/xml,text/html,text/xml,text/plain
spring.main.banner-mode=off
spring.mvc.view.prefix=/WEB-INF/jsp/
spring.mvc.view.suffix=.jsp
```

**Comment by bmcd87** [comment]
Since you're not having cakeshop run a node and are instead connecting to your existing 5 node network, you don't need to worry about those values. It looks like you have auto start and stop set to false, which is good.

You should be able to run cakeshop, then open it up in the browser. From there, choose Manage Nodes from the dropdown in the top right hand corner and add each of the 5 nodes from that page.

We're working on greatly simplifying the configuration of cakeshop to  make this much clearer. Let me know if there are any other issues!

**Comment by vasa-develop** [comment]
Ok. I started and added 5 nodes.
When I check the status of Node 1 (actually, any node)
I'm getting the Blocks/per minute correctly, but the Node status shows "Down".

What could I have done wrong here?

**Comment by bmcd87** [comment]
Hmm, never seen that before. Can you expand the Node Info widget and take a screenshot of the page? Also go to the Peers tab, does the peers list contain all 5 nodes? Any errors in the javascript console?

Are you able to connect and interact with the network using `geth attach` in your terminal?

**Comment by vasa-develop** [comment]
Cool. I'll try this and get back to you :)

**Comment by vasa-develop** [comment]
Hi @bmcd87! Sorry for the late reply

**TLDR; Scroll down to see console outputs.**

> Can you expand the Node Info widget and take a screenshot of the page? 

Not sure which widget are we talking about here, but the "manage nodes" drop-down does show me the 5 nodes I have added.

Also, a weird thing that happens when I start the console for the first time, the number of blocks/min keeps rising, until I refresh the page. Also, trx/min rises for a while, and then maintains a static tpm.

![console](https://user-images.githubusercontent.com/28847087/78247867-402bf700-7509-11ea-8790-5312d4cc20a1.png)

> Also go to the Peers tab, does the peers list contain all 5 nodes?

Nope. it's empty.

![peer1](https://user-images.githubusercontent.com/28847087/78247790-1c68b100-7509-11ea-8243-4bfbd6334a02.png)

> Are you able to connect and interact with the network using geth attach in your terminal?

Yup. I can see all the 5 nodes are connected to each other.

> Any errors in the javascript console?

YES. A couple of them.

![err1](https://user-images.githubusercontent.com/28847087/78248023-7e291b00-7509-11ea-9423-418011ed96a4.png)
![err2](https://user-images.githubusercontent.com/28847087/78248030-81bca200-7509-11ea-90a4-592df5a2d83d.png)

**Comment by bmcd87** [comment]
Can you share the command you use to start your geth nodes?

**Comment by vasa-develop** [comment]
Here you go!

```
#!/bin/bash
PRIVATE_CONFIG=ignore nohup geth --datadir data --nodiscover --istanbul.blockperiod 1 --syncmode full --mine --minerthreads 1 --verbosity 5 --networkid 10 --rpc --rpccorsdomain "*" --rpcaddr 0.0.0.0 --rpcport 22000 --rpcapi admin,db,eth,debug,miner,net,shh,txpool,personal,web3,quorum,istanbul --emitcheckpoints --port 30303 2>>node.log &
```

I have identical scripts for all 5 nodes.

**Comment by bmcd87** [comment]
Hmm, seems fine. Can you look at the cakeshop logs in the terminal (java logs, not the developer console), then change the node in the dropdown menu and see what errors show up in the logs when it tries to connect to the new node?

**Comment by vasa-develop** [comment]
The cakeshop is running on Node 1. Initially, the dashboard was showing Node 1 stats, after a while I switched the dashboard to Node 2.

[Here are the logs](https://pastebin.com/A5ZNGg6m)

As you can see in the following image, which shows the visual pattern in the logs, you can see a pattern in followed in the start (while dashboard shows Node 1), and the end (while dashboard shows Node 2). But in the middle, it doesn't follow the pattern (which are the logs from the moment when I transitioned the dashboard from Node 1 to Node 2).

![code](https://user-images.githubusercontent.com/28847087/78452246-dac23c80-76a7-11ea-8394-4e029093cf97.png)

**Comment by vasa-develop** [comment]
@bmcd87 I can share the URL to the dashboard with you privately, in case that would help in debugging this.

**Comment by bmcd87** [comment]
Seems like there might be something wrong with the node urls. Can you show the manage nodes screen? 

Also, if you want to join our Slack channel we can help there too: https://bit.ly/quorum-slack

**Comment by vasa-develop** [comment]
@bmcd87 Sure, let's continue this over the Slack channel
